### PR TITLE
Billboard activity implementation

### LINKED ISSUE #14: Film selection activity
Allows users to select a film before they can start the booking process

### PR DESCRIPTION
**Pull request by alexaib2002**
- Implements #14 

## TODO
- @alexeses should mark this PR as ready once the requirements are finished.

### CODE REVIEW

**Comment by aleexgheorghe** [comment]
Hello team!

The first version of the movie selection banner is finished, I have left everything ready so that our colleague @alexaib2002 can easily implement the data movement between activities. 

- [ ] It remains to get all the movies data from the database.

Interface images:
image


Thank you.

**Comment by alexaib2002** [comment]
Good work! I'll mark this PR as ready for review, and as soon as it's merged, I'll start the activity transitions.

File: app/CineHub/app/src/main/java/org/cinehub/BillBoardActivity.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.os.Bundle;
+
+import org.cinehub.utils.MovieAdapter;
+
+public class BillBoardActivity extends AppCompatActivity {
+
+    private MovieAdapter adapter;
+    private RecyclerView rv_movies;
```
Comment (alexaib2002) [inline app/CineHub/app/src/main/java/org/cinehub/BillBoardActivity.java]: This variable should be written in camelCase as it's a Java var.

File: app/CineHub/app/src/main/res/layout/item_movie.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                    android:text="Exanple Movie Title"
+                    android:textColor="#0B090A"
+                    android:textSize="20sp"
+                    android:textStyle="bold" />
+
+                <TextView
+                    android:id="@+id/tvMovieDescription"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:text="This is a description of the movie that is being shown in the billboard activity aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa."
```
Comment (alexaib2002) [inline app/CineHub/app/src/main/res/layout/item_movie.xml]: Hardcoded text, should be specified as a placeholder on strings.xml

File: app/CineHub/app/src/main/java/org/cinehub/utils/MovieAdapter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public int getItemCount() {
+        return 0;
+        // Return the number of movies API
+    }
+
+
+    public static class MovieVH extends RecyclerView.ViewHolder {
+
+        TextView tv_title, tv_release_date, tv_description;
```
Comment (alexaib2002) [inline app/CineHub/app/src/main/java/org/cinehub/utils/MovieAdapter.java]: Variable names should be written in camelCase

File: app/CineHub/app/src/main/java/org/cinehub/utils/MovieAdapter.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import org.cinehub.R;
+
+import java.util.List;
+
+public class MovieAdapter extends RecyclerView.Adapter<MovieAdapter.MovieVH> {
```
Comment (alexaib2002) [inline app/CineHub/app/src/main/java/org/cinehub/utils/MovieAdapter.java:18]: Has this adapter been implemented?

**Review by alexaib2002 — COMMENTED** [review]
Nice code overall, but several variables are written in snake_case instead of camelCase.
The RecyclerView adapter is not implemented, and while this shouldn't be a merge-blocking issue, I think it would be nice to have it finished before it arrives to the `main` branch.
Strings of `item_movie` are hardcoded; if we want to keep them as placeholders, they should be specified as `placeholder_*` in the `strings.xml` file

File: app/CineHub/app/src/main/res/layout/item_movie.xml (hunk 2 above)
Comment (aleexgheorghe) [reply app/CineHub/app/src/main/res/layout/item_movie.xml]: Thanks! It was an oversight, I was seeing how the design looked like.

File: app/CineHub/app/src/main/java/org/cinehub/BillBoardActivity.java (hunk 1 above)
Comment (aleexgheorghe) [reply app/CineHub/app/src/main/java/org/cinehub/BillBoardActivity.java]: Fix!

**Comment by alexaib2002** [comment]
This PR won't be required anymore; #38 has superseded it and also implements extended functionality

**Pull request closed without merging**